### PR TITLE
feat(money): pin confirmation functionality

### PR DESCRIPTION
**Pull request by JorgeDelgadillo**
[TAD](https://instawork.atlassian.net/wiki/spaces/EN/pages/2828468280/TAD+IW+Card+Pin+Confirmation)

# Summary 

Added the function `confirmPin` on `Swift` and `Kotlin` that compares the pin and pinConfirm values

### CODE REVIEW

File: android/src/main/java/com/vgscollectreactnative/VgsCollectReactNativeViewManager.kt [hunk 1]
```diff
@@ -1,11 +1,13 @@
 package com.vgscollectreactnative
 
+import android.annotation.SuppressLint
```
Comment (tecnoloco) [inline android/src/main/java/com/vgscollectreactnative/VgsCollectReactNativeViewManager.kt]: I would suggest against using suppresslint to remove lint warnings

File: ios/CollectorManager.swift [hunk 2]
```diff
@@ -15,6 +15,30 @@ class CollectorManager: NSObject {
...
+  
+    @objc(pinConfirm:path:method:headers:resolver:rejecter:)
+    func pinConfirm(_ name: String, path: String, method: String, headers: Dictionary<String, String>, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) -> Bool {
+      let collector = CollectorManager.map[name]!
+        
+      if (collector == nil) {
+          reject("error", "invalid name", NSError(domain: "MissingCollectorByNameError", code: 401))
+          return;
+      }
+      collector.customHeaders = headers;
```
Comment (tecnoloco) [inline ios/CollectorManager.swift]: I don't think we will be using headers for pin confitmation.

File: ios/CollectorManager.swift [hunk 3]
```diff
@@ -15,6 +15,30 @@ class CollectorManager: NSObject {
         
     CollectorManager.map[name] = VGSCollect(id: vaultId, environment: environment);
  }
+  
+    @objc(pinConfirm:path:method:headers:resolver:rejecter:)
+    func pinConfirm(_ name: String, path: String, method: String, headers: Dictionary<String, String>, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) -> Bool {
```
Comment (tecnoloco) [inline ios/CollectorManager.swift]: Can we revisit which parameters are really being used/needed in pinConfirm?

Comment (tecnoloco) [reply ios/CollectorManager.swift]: I would say that it needs the filed names to be compared

File: ios/CollectorManager.swift [hunk 4]
```diff
@@ -15,6 +15,30 @@ class CollectorManager: NSObject {
...
+      let collector = CollectorManager.map[name]!
+        
+      if (collector == nil) {
+          reject("error", "invalid name", NSError(domain: "MissingCollectorByNameError", code: 401))
+          return;
+      }
+      collector.customHeaders = headers;
+
+      let pin = collector.getTextField(fieldName: "data.attributes.pin")
```
Comment (tecnoloco) [inline ios/CollectorManager.swift:27]: I would say that field names should be dynamic and not a fixed value

File: ios/CollectorManagerBridge.m [hunk 5]
```diff
@@ -11,5 +11,14 @@ @interface RCT_EXTERN_MODULE(CollectorManager, NSObject)
...
                   rejecter:(RCTPromiseRejectBlock)reject
                   )
+                  
+RCT_EXTERN_METHOD(pinConfirm:(NSString *)name
+                  path:(NSString *)path
+                  method:(NSString *)method
+                  headers:(NSDictionary *)headers
+                  resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject
+                  )
```
Comment (tecnoloco) [inline ios/CollectorManagerBridge.m]: This interface should change after revisiting the parameters that are used /needed here

File: android/src/main/java/com/vgscollectreactnative/CollectorManager.kt [hunk 6]
```diff
@@ -26,6 +29,26 @@ class CollectorManager internal constructor(context: ReactApplicationContext?) :
     }
   }
 
+  @SuppressLint("ResourceType")
```
Comment (tecnoloco) [inline android/src/main/java/com/vgscollectreactnative/CollectorManager.kt]: I wouldn't suggest using lint suppression.

File: android/src/main/java/com/vgscollectreactnative/CollectorManager.kt [hunk 7]
```diff
@@ -26,6 +29,26 @@ class CollectorManager internal constructor(context: ReactApplicationContext?) :
...
+  @ReactMethod
+  fun confirmPin(name: String, path: String, method: String, headers: ReadableMap, callback: Callback) {
```
Comment (tecnoloco) [inline android/src/main/java/com/vgscollectreactnative/CollectorManager.kt]: here, I would revisit parameters being used/needed for pinConirm. Also, I would use the method name for ios and android: here is `confirmPin` and ios is `pinConfirm`

**Comment by tecnoloco** [comment]
@JorgeDelgadillo can you add TAD link to the description for more context.

File: android/src/main/java/com/vgscollectreactnative/VgsCollectReactNativeViewManager.kt (hunk 1 above)
Comment (JorgeDelgadillo) [reply android/src/main/java/com/vgscollectreactnative/VgsCollectReactNativeViewManager.kt]: Done [here](https://github.com/Instawork/vgs-collect-react-native/pull/1/commits/65850574faec0cae00f0d2b793cf1529b5161f2f#diff-a41cc746377eae3513fca397551becca1eaaf1350440859511e60f1cb4dec243L3)

File: ios/CollectorManager.swift (hunk 2 above)
Comment (JorgeDelgadillo) [reply ios/CollectorManager.swift]: Thanks, i removed the parameters that we don't use [here](https://github.com/Instawork/vgs-collect-react-native/pull/1/commits/65850574faec0cae00f0d2b793cf1529b5161f2f#diff-a41cc746377eae3513fca397551becca1eaaf1350440859511e60f1cb4dec243L34)

File: ios/CollectorManagerBridge.m (hunk 5 above)
Comment (JorgeDelgadillo) [reply ios/CollectorManagerBridge.m]: Done [here](https://github.com/Instawork/vgs-collect-react-native/pull/1/commits/65850574faec0cae00f0d2b793cf1529b5161f2f#diff-c10ccafa7325ad5fda7bcf295f9ec21845719860ce5973124b7ffa00e40b8bd5L16)

File: ios/CollectorManager.swift (hunk 4 above)
Comment (tecnoloco) [reply ios/CollectorManager.swift:27]: @JorgeDelgadillo Just take into account this comment [here](https://github.com/Instawork/mobile/pull/4865#discussion_r1106123340)

**Review by tecnoloco — APPROVED** [review]
I would take into account this comment [here](https://github.com/Instawork/mobile/pull/4865#discussion_r1106123340)

Other than that LGTM overall

**Pull request merged**